tictactoe.py

```diff
@@ -0,0 +1,202 @@
+import math
+import random
+ 
+ 
+class TicTacToe:
+    """
+    a class for the TicTacToe game
+    """
+    def __init__(self, board_size = 3, win_length = 3, ai = True):
+        """
+        None for empty
+        True for O
+        False for X
+        """
+        self.board_size = board_size
+        self.num_positions = board_size * board_size
+        self.board = [None] * self.num_positions
+        self.win_length = win_length
+        # ai side
+        self.ai = ai
+ 
+    def display(self, input_list):
+        assert len(input_list) == self.num_positions
+        for row in xrange(self.board_size):
+            start_pos = row * self.board_size
+            end_pos = start_pos + self.board_size
+            row_str = '|'.join(map(str, input_list[start_pos:end_pos]))
+            print row_str
+ 
+    def display_movement(self):
+        print "Movement Key:"
+        self.display(range(1, self.num_positions+1))
+ 
+    def display_board(self):
+        print "Board:"
+        input = [' ' if piece is None else 'O' if piece == True else 'X'
```

Comment:
A list comprehension over two lines is not a good idea, use a for loop.

`piece == True` should be removed in favor of `if piece`

Refer to http://stackoverflow.com/questions/9494404/use-of-true-false-and-none-as-return-values-in-python-functions